### DISCUSSION TITLE
Questions regarding the diskStorage cache

### DISCUSSION THREAD

**Discussion by FaizanDurrani** [onevcat/Kingfisher#2096] [Q&A] [answered]
Hello,

I am trying to implement a semi-persistent cache for Kingfisher.

I've observed that the default disk cache url points to the `.cachesDirectory` which the OS can clear based on storage pressure.

My question is, what are the consequences of changing the disk cache url to point to `.documentDirectory` such that iOS doesn't clear it based on the storage remaining?

I am aware that this significantly increases the size of the document directory but apart from that, is there any restrictions on Kingfisher's end?

~~I have testers reporting that their image cache is being cleared between app restarts even when the url points to the document folder, is there something that I should be doing along with setting `directory` in `Config` initialiser?~~ apparently I had legacy code that was interfering with the `sizeLimit` causing the cache to clear due to exceeding size limit. The other questions still apply though.

**Comment by onevcat** [comment, marked as answer]
There should be no restriction from Kingfisher to write the cached images to the `.documentDirectory`. However, it apparently violates Apple's guideline of "putting the re-generatable contents to cache folder". I am not sure if Apple would reject this in the review or not, so use it at your risk.

Besides of the `sizeLimit`, there is also an expiration duration inside Kingfisher to clear outdated cached images from time to time. Refer to [the wiki](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#set-default-expiration-for-cache) for more. Basically, you may want to also set it:

```swift
cache.diskStorage.config.expiration = .never
```

**Reply by FaizanDurrani** [reply]
> However, it apparently violates Apple's guideline of "putting the re-generatable contents to cache folder". I am not sure if Apple would reject this in the review or not, so use it at your risk.

Hmm, I am using Kingfisher diskStorage as an offline store for the thumbnails for items in the database, such that if you open up the app without an active internet connection there's a good chance that the images are still available without relying on iOS to not clear the cache.

I believe this is a good enough reason to switch from the temporary cache folder (as it's no different from just downloading the thumbnails for items in the db).
